eng/docgeneration/Generate-DocIndex.ps1

```diff
@@ -12,153 +27,95 @@ Write-Verbose "Initializing Default DocFx Site..."
 & "${DocFxTool}" init -q -o "${DocOutDir}"
 
 Write-Verbose "Copying template and configuration..."
-New-Item -Path "${DocOutDir}" -Name "templates" -ItemType "directory"
+New-Item -Path "${DocOutDir}" -Name "templates" -ItemType "directory" -Force
 Copy-Item "${DocGenDir}/templates/*" -Destination "${DocOutDir}/templates" -Force -Recurse
 Copy-Item "${DocGenDir}/docfx.json" -Destination "${DocOutDir}/" -Force
-
-Write-Verbose "Creating Index using service directory and package names from repo..."
-# The service mapper is used to map the directory names to the service names to produce
-# a more friendly index. If something isn't in the mapper then the default will just be
-# the service name in all caps
-$serviceMapHash = Get-Content -Path "${DocGenDir}/service-mapper.json" | ConvertFrom-Json -AsHashtable
-
-# There are some artifact that show up, due to the way we do discovery, that are never shipped.
-# Keep a list of those here and anything we don't want to ship can be added to here which will
-# cause them to get skipped when generating the DocIndex
-$ArtifactsToSkip = (
-'azure-cosmos-benchmark',
-'azure-sdk-template'
-)
-
-# The list of services is being constructed from the directory list under the sdk folder
-# which, right now, only contains client/data directories. When management is moved to
-# the under sdk it'll automatically get picked up.
-$ServiceListData = Get-ChildItem "${RepoRoot}/sdk" -Directory
 $YmlPath = "${DocOutDir}/api"
 New-Item -Path $YmlPath -Name "toc.yml" -Force
-foreach ($Dir in $ServiceListData)
-{   
-    $mappedDir = ""
-    if ($serviceMapHash.ContainsKey($Dir.Name)) 
-    {
-        $mappedDir = $serviceMapHash[$Dir.Name]
-    }
-    else
-    {
-        $mappedDir = $Dir.Name.ToUpper()
-    }
-
-    # Store the list of artifacts into the arrays and write them into the .md file
-    # after processing the list of subdirectories. This will allow the correct 
-    # division of the artifacts under the Client or Management headings
-    $clientArr = @()
-    $mgmtArr = @()
 
-    $PkgList = Get-ChildItem $Dir.FullName -Directory -Exclude changelog, faq, .github, build
-    if (($PkgList | Measure-Object).count -eq 0)
-    {
-        continue
+Write-Verbose "Reading artifact from storage blob ..."
+$metadata = GetMetaData -lang $lang
+$langRegex = $LangMapping[$lang][1]
+$titleRegex = $LangMapping[$lang][0]
+$regex = "^$langRegex/(.*)/$"
+$pageToken = ""
+# Used for sorting the toc display order
+$orderSet = @{}
+$orderArray = @()
```

Comment:
Why do we need the orderArray? it seems like we could just sort the orderSet.Key if you want to sort.